website/docs/04_design/package-config.md

```diff
@@ -0,0 +1,160 @@
+# Package Configuration
+
+:::info Disclaimer
+This is a design document.
+Some or all of the functionality described in the following may not be implemented in a released version of Glasskube.
+:::
+
+## Probelm analysis
+
+At Glasskube, what we mean when we talk about package configuration, is a controlled alteration of the resources that are part of a package depending on values specified by the user.
+This can be thought of as three sub-problems:
+
+### What values are available for a given package
+
+The author of a package may include a declaration of the possible configuration items in the package manifest, where each package may have multiple **value specifications**.
+A value specification contains information to help clients display an appropriate form item for data entry and constraints to be validated by the client as well as the package operator.
+
+### How does a value affect the deployed package resources
+
+Additionally, a value specification describes some alterations to the deployed package.
+We call those alterations **targets**.
+Each value specification may contain a number of targets.
+A target can be one of two things:
+
+1. A change to the default values of a helm release contained in the package
+2. A JSON patch that should be applied to a resource contained in a `manifests` entry of the package
+
+### How does a value affect the deployed package resources
+
+A user, when installing a package using Glasskube, may declare a **value** for each value specification of that package.
+A value can be either a **literal value** or a **reference value**.
+Literal values represent a simple value (e.g. a string entered via a text field).
+Reference values represent references to values in other resources in the same Kubernetes cluster.
+Such references can be secrets, configmaps or other packages.
+
+## Design proposal
+
+The `PackageManifest` has a property `Values` of type `map[string]ValueSpecification`.
+The key in this map is referred to as that values **name**
+`ValueSpecification` is a struct with the following properties:
+
+- **`Type`** (`string` enum):
+  Every value must have a type, so that we know what kind of input field to show for this value.
+  Initially, this can be one of `check`, `options`, `text`, `number` but it is possible to add more types in future releases.
+- **`Label`** (`string`):
+  The label is used to denote an input field related to this value in a UI.
+  Specifying a label is optional, by default the name of the value should be used.
+- **`Description`** (`string`):
+  The description can be used to give more context to a value. Also optional.
+- **`DefaultValue`** (`string`):
+  The default value is pre-selected/pre-filled in the form field of this value for new packages.
+- **`Options`** (`[]string`):
+  Available choices for values of type options.
+  Should be ignored for other types.
+- **`Hints`** (`string` enum):
```

Comment:
Also I want to offer my 2 cents on UI form configurability (`Hints`), speaking from past experiences: This can (and probably will) become very messy. 
I can already see that once we can render a list of form fields with these basic configuration options, it will look too basic. So we will come up with optional options like grouping fields together, showing/hiding/filling fields based on the value of other fields, adding/removing constraints from fields based on the value of other fields, and so on and so forth. Once we allow fields to be dependent on each other, it will get very tricky. Not to forget that this also must be considered on the CLI. Long story short, it's a minefield. 
I think the challenge here will become to separate actual package configuration from some advanced UI form configurability. And we should try to not compromise an understandable package configuration for the sake of some fancy UI features. 

Not saying we will be able to prevent this mess forever – but you heard it here first :)